bayesian/api/api_v2.py

```diff
@@ -243,13 +244,20 @@ def stack_analyses():
         sa_post_request = StackAnalysesPostRequest(**request.form, **request.files)
 
     except ValidationError as e:
-        # 2. Check of invalid params and raise exception.
+        # Check of invalid params and raise exception.
         error_message = 'Validation error(s) in the request.'
         for error in e.errors():
             error_message += ' {} => {}.'.format(error['loc'][0], error['msg'])
         logger.exception(error_message)
         raise HTTPError(400, error=error_message) from e
 
+    # 2. Check if ecosystem and manifest file name match
+    ecosystem_from_manifest = get_ecosystem_from_manifest(sa_post_request.manifest.filename)
+    if sa_post_request.ecosystem != ecosystem_from_manifest:
+        raise HTTPError(400, 'Error processing request. Manifest {} and ecosystem {} does '
+                             'not match'.format(sa_post_request.manifest.filename,
+                                                sa_post_request.ecosystem))
+
```

Comment:
This validation can be part of sa_models right?

Reply:
Here we want to verify value of two parameter with respect to each other. Can you point me how this can be achieved using pydantic? (I tried initially,but could not find working solution)

Reply:
Try with root validator.

https://pydantic-docs.helpmanual.io/usage/validators/#root-validators